src/hooks/useConnectWallet.ts

```diff
@@ -127,16 +134,21 @@ export const useConnectWallet = () => {
 
   watch(
     [isConnectedNetwork],
-    () => {
+    async () => {
       const address = localStorage.getItem(SELECTED_ADDRESS);
-      if (!address || !isConnectedNetwork.value) return;
-      // Memo: wait for updating the chain id from the initial state 592 (to pass the `setupNetwork` function)
-      setTimeout(async () => {
-        if (address === 'Ethereum Extension') {
+      const isMetaMaskDeepLink = window.location.hash === deepLinkPath.metamask;
```

Comment:
Can you add deep links for other mobile wallets too? Like Math wallet and Nova wallet

Reply:
I've checked with MathWallet and confirmed MathWallet doesn't have this feature at this moment.